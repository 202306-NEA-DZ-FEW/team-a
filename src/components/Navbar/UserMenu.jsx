import { Menu, Transition } from "@headlessui/react";
import Image from "next/image";
import Link from "next/link";
import { Fragment } from "react";
import { BiLogOut } from "react-icons/bi";
import { MdOutlineDashboardCustomize } from "react-icons/md";

function UserMenu({ user, logOut, t, userProfile }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }

  return (
    <Menu as='div' className='relative hidden lg:block'>
      <Menu.Button className='btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost'>
        <div className='avatar'>
          <div className='w-8 rounded-full ring ring-primary hover:ring-offset-1 transition-all duration-500 ring-offset-base-100 ring-offset-2'>
            <Image width={100} height={100} alt='user' src={userProfile} />
          </div>
        </div>
      </Menu.Button>
      <Transition
        as={Fragment}
        enter='transition ease-out duration-100'
        enterFrom='transform opacity-0 scale-95'
        enterTo='transform opacity-100 scale-100'
        leave='transition ease-in duration-75'
        leaveFrom='transform opacity-100 scale-100'
        leaveTo='transform opacity-0 scale-95'
      >
        <Menu.Items className='absolute right-0 translate-x-1/3 left-auto origin-center z-40 top-12 lg:mr-4 w-screen md:w-screen lg:w-56 rounded-xl p-2 bg-base-100 shadow-xl ring-1 ring-black ring-opacity-5 focus:outline-none font-light tracking-wider'>
          <Menu.Item>
            {({ active }) => (
              <Link
                className={classNames(
                  active ? "bg-base-300" : "text-gray-700",
                  "px-4 py-2 text-sm mb-2 rounded-md w-full flex gap-1 items-center"
                )}
                href={{
                  pathname: "/dashboard",
                  query: { user: user.uid },
                }}
              >
                <MdOutlineDashboardCustomize size={20} />
                {t("common:navbar:dashboard")}
              </Link>
            )}
          </Menu.Item>
          <Menu.Item>
            {({ active }) => (
              <button
                className={classNames(
                  active ? "bg-base-300" : "text-gray-700",
                  "px-4 py-2 text-sm mb-2 rounded-md w-full flex gap-1 items-center"
                )}
                onClick={logOut}
              >
                <BiLogOut size={20} />
                {t("common:buttons:signOut")}
              </button>
            )}
          </Menu.Item>
        </Menu.Items>
      </Transition>
    </Menu>
  );
}

export default UserMenu;
