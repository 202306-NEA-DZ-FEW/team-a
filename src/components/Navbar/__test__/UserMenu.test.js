import renderer from "react-test-renderer";

import UserMenu from "../UserMenu";

it("it renders correctly", () => {
  const mockT = jest.fn();
  const mockLogOut = jest.fn();
  const tree = renderer
    .create(<UserMenu logOut={mockLogOut} user={{}} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
