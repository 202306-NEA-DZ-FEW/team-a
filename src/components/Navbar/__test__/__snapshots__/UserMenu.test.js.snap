// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="relative hidden lg:block"
  data-headlessui-state=""
>
  <button
    aria-expanded={false}
    aria-haspopup="menu"
    className="btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost"
    data-headlessui-state=""
    id="headlessui-menu-button-:r0:"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <div
      className="avatar"
    >
      <div
        className="w-8 rounded-full ring ring-primary hover:ring-offset-1 transition-all duration-500 ring-offset-base-100 ring-offset-2"
      >
        <img
          alt="user"
          data-nimg="1"
          decoding="async"
          height={100}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={100}
        />
      </div>
    </div>
  </button>
</div>
`;
